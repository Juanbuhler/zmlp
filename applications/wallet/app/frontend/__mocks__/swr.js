import { createElement } from 'react'

const noop = () => () => {}

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * useSWR
 */

let mockUseSWRResponse = {}

export const __setMockUseSWRResponse = data => {
  mockUseSWRResponse = { revalidate: noop, ...data }
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
