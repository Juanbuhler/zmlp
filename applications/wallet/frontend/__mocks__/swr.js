import { createElement } from 'react'

const noop = () => () => {}

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * cache
 */

export const { cache } = jest.requireActual('swr')

/**
 * useSWR
 */

let mockUseSWRResponse = { revalidate: noop, mutate: noop }

export const __setMockUseSWRResponse = (data) => {
  mockUseSWRResponse = { revalidate: noop, mutate: noop, ...data }
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
