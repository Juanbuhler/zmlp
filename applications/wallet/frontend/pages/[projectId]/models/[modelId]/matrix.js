import ModelMatrix from '../../../../src/ModelMatrix'

export default ModelMatrix
