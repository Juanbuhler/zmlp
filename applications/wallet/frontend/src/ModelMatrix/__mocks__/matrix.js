const matrix = {
  name: 'flowers',
  datasetId: '4b0b10a8-cec1-155c-b12f-ee2bc8787e06',
  labels: ['pink', 'red', 'yellow'],
  matrix: [
    [0, 0, 2],
    [1, 1, 0],
    [0, 0, 0],
  ],
  maxScore: 1,
  minScore: 0,
  moduleName: 'flower-types',
  overallAccuracy: 0.6,
  testSetOnly: true,
  isMatrixApplicable: true,
}

export default matrix
