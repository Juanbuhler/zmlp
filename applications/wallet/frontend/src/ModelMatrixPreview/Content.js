import useSWR from 'swr'
import PropTypes from 'prop-types'

import { colors, spacing } from '../Styles'

import { useLocalStorage } from '../LocalStorage/helpers'
import { reducer } from '../Resizeable/reducer'

import { PANEL_WIDTH } from '../ModelMatrix/helpers'

const FROM = 0
const SIZE = 28

const ModelMatrixPreviewContent = ({ encodedFilter, projectId }) => {
  const {
    data: { results, count },
  } = useSWR(
    `/api/v1/projects/${projectId}/searches/query/?query=${encodedFilter}&from=${FROM}&size=${SIZE}`,
  )

  const [{ size }] = useLocalStorage({
    key: `Resizeable.ModelMatrixPreview`,
    reducer,
    initialState: {
      size: PANEL_WIDTH,
      originSize: 0,
      isOpen: false,
    },
  })

  return (
    <div
      css={{
        flex: 1,
        // height: '100%',
        height: '0%',
        backgroundColor: colors.structure.coal,
        padding: spacing.small,
        overflow: 'auto',
      }}
    >
      <div
        css={{
          display: 'grid',
          gridTemplateColumns: `repeat(${Math.floor(
            size / 200,
          )}, minmax(200px, 1fr))`,
          gap: spacing.small,
        }}
      >
        {results.map(({ thumbnailUrl, metadata, id }) => {
          const { pathname: thumbnailSrc } = new URL(thumbnailUrl)

          return (
            <div
              key={id}
              css={{
                position: 'relative',
                backgroundColor: colors.structure.mattGrey,
                aspectRatio: '1 / 1',
              }}
            >
              <img
                css={{
                  position: 'absolute',
                  maxWidth: '100%',
                  maxHeight: '100%',
                  top: '50%',
                  left: '50%',
                  transform: 'translate(-50%, -50%)',
                }}
                src={thumbnailSrc}
                alt={metadata?.source?.filename}
              />
            </div>
          )
        })}
      </div>
    </div>
  )
}

ModelMatrixPreviewContent.propTypes = {
  encodedFilter: PropTypes.string.isRequired,
  projectId: PropTypes.string.isRequired,
}

export default ModelMatrixPreviewContent
