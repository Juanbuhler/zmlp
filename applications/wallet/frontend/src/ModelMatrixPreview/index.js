import { useRouter } from 'next/router'
import Link from 'next/link'
import PropTypes from 'prop-types'

import { colors, constants, spacing, typography } from '../Styles'

import { useLocalStorage } from '../LocalStorage/helpers'

import FilterSvg from '../Icons/filter.svg'

import Button, { VARIANTS as BUTTON_VARIANTS } from '../Button'
import { MIN_WIDTH as PANEL_MIN_WIDTH } from '../Panel'
import { ACTIONS, reducer as resizeableReducer } from '../Resizeable/reducer'

import SuspenseBoundary from '../SuspenseBoundary'

import { encode } from '../Filters/helpers'

import ModelMatrixPreviewContent from './Content'

const ModelMatrixPreview = ({
  settings: { selectedCell, minScore, maxScore },
  labels,
  moduleName,
}) => {
  const {
    query: { projectId, modelId },
  } = useRouter()

  const [, setRightOpeningPanel] = useLocalStorage({
    key: 'rightOpeningPanelSettings',
    reducer: resizeableReducer,
    initialState: {
      size: PANEL_MIN_WIDTH,
      originSize: 0,
      isOpen: false,
    },
  })

  if (!selectedCell.length) {
    return (
      <div
        css={{
          padding: spacing.normal,
          fontStyle: typography.style.italic,
          height: '100%',
          borderLeft: constants.borders.regular.coal,
        }}
      >
        Select a cell of the matrix to view the labeled assets.
      </div>
    )
  }

  const encodedFilter = encode({
    filters: [
      {
        type: 'label',
        attribute: `labels.${moduleName}`,
        modelId,
        values: {
          labels: [labels[selectedCell[0]]],
          scope: 'test',
        },
      },
      {
        attribute: `analysis.${moduleName}`,
        isDisabled: false,
        type: 'labelConfidence',
        values: {
          labels: [labels[selectedCell[1]]],
          min: minScore,
          max: maxScore,
        },
      },
    ],
  })

  return (
    <div
      css={{
        flex: 1,
        display: 'flex',
        flexDirection: 'column',
        borderLeft: constants.borders.regular.coal,
        overflow: 'hidden',
      }}
    >
      <div
        css={{
          padding: spacing.normal,
          fontSize: typography.size.regular,
          lineHeight: typography.height.regular,
        }}
      >
        <Link href={`/${projectId}/visualizer?query=${encodedFilter}`} passHref>
          <Button
            aria-label="View Filter Panel"
            variant={BUTTON_VARIANTS.SECONDARY_SMALL}
            onClick={() => {
              setRightOpeningPanel({
                type: ACTIONS.OPEN,
                payload: { openPanel: 'filters' },
              })
            }}
            style={{
              display: 'flex',
              paddingTop: spacing.moderate,
              paddingBottom: spacing.moderate,
            }}
          >
            <div css={{ display: 'flex', alignItems: 'center' }}>
              <FilterSvg
                height={constants.icons.regular}
                css={{ paddingRight: spacing.base }}
              />
              View Filter Panel
            </div>
          </Button>
        </Link>

        <div css={{ height: spacing.normal }} />

        <h3>
          <span
            css={{
              fontFamily: typography.family.condensed,
              fontWeight: typography.weight.regular,
              color: colors.structure.zinc,
            }}
          >
            True:
          </span>{' '}
          {labels[selectedCell[0]]}
        </h3>
        <h3>
          <span
            css={{
              fontFamily: typography.family.condensed,
              fontWeight: typography.weight.regular,
              color: colors.structure.zinc,
            }}
          >
            Pred:
          </span>{' '}
          {labels[selectedCell[1]]}
        </h3>
      </div>

      <SuspenseBoundary>
        <ModelMatrixPreviewContent
          encodedFilter={encodedFilter}
          projectId={projectId}
        />
      </SuspenseBoundary>
    </div>
  )
}

ModelMatrixPreview.propTypes = {
  settings: PropTypes.shape({
    selectedCell: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
    minScore: PropTypes.number.isRequired,
    maxScore: PropTypes.number.isRequired,
  }).isRequired,
  labels: PropTypes.arrayOf(PropTypes.string).isRequired,
  moduleName: PropTypes.string.isRequired,
}

export default ModelMatrixPreview
