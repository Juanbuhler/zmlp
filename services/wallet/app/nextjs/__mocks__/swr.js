import { createElement } from 'react'

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * useSWR
 */

let mockUseSWRResponse = {}

export const __setMockUseSWRResponse = data => {
  mockUseSWRResponse = data
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
