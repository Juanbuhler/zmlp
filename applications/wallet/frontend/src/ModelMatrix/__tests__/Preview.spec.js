import TestRenderer, { act } from 'react-test-renderer'

import matrix from '../__mocks__/matrix'

import ModelMatrixPreview from '../Preview'

describe('<ModelMatrixPreview />', () => {
  it('should render properly', () => {
    require('swr').__setMockUseSWRResponse({
      data: matrix,
    })

    const component = TestRenderer.create(
      <ModelMatrixPreview matrix={matrix} selectedCell={[0, 1]} />,
    )

    act(() => {
      component.root
        .findByProps({ 'aria-label': 'View Filter Panel' })
        .props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly with no selected cells', () => {
    require('swr').__setMockUseSWRResponse({
      data: matrix,
    })

    const component = TestRenderer.create(
      <ModelMatrixPreview matrix={matrix} selectedCell={[]} />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
