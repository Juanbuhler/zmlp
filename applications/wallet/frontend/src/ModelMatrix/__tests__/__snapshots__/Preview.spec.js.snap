// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModelMatrixPreview /> should render properly 1`] = `
Array [
  .emotion-0 {
  padding: 16px;
  font-size: 15px;
  line-height: 19px;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 8px 16px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding-top: 12px;
  padding-bottom: 12px;
}

.emotion-1,
.emotion-1:hover,
.emotion-1:visited,
.emotion-1 .noop {
  background-color: #808080;
}

.emotion-1:hover,
.emotion-1 .noop {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #b3b3b3;
}

.emotion-1[aria-disabled=true],
.emotion-1 .noop {
  color: #2e2e2e;
  background-color: #808080;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  padding-right: 8px;
}

.emotion-4 {
  height: 16px;
}

.emotion-5 {
  font-family: Roboto Condensed;
  font-weight: 400;
  color: #b3b3b3;
}

<div
    className="emotion-0"
  >
    <button
      aria-label="View Filter Panel"
      className="emotion-1"
      onClick={[Function]}
      type="button"
    >
      <div
        className="emotion-2"
      >
        <svg
          className="emotion-3"
          data-file-name="filter.svg"
          height={20}
        />
        View Filter Panel
      </div>
    </button>
    <div
      className="emotion-4"
    />
    <h3>
      <span
        className="emotion-5"
      >
        True:
      </span>
       
      Dahlia
    </h3>
    <h3>
      <span
        className="emotion-5"
      >
        Pred:
      </span>
       
      Cherry Blossom
    </h3>
  </div>,
  .emotion-0 {
  background-color: #191919;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
    className="emotion-0"
  />,
]
`;

exports[`<ModelMatrixPreview /> should render properly with no selected cells 1`] = `
.emotion-0 {
  padding: 16px;
  font-style: italic;
}

<div
  className="emotion-0"
>
  Select a cell of the matrix to view the labeled assets.
</div>
`;
